kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/ScoreWrapper.java

```diff
@@ -0,0 +1,107 @@
+package org.kie.server.api.model.instance;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlValue;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamConverter;
+import com.thoughtworks.xstream.converters.extended.ToAttributedValueConverter;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.buildin.bendable.BendableScore;
+import org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScore;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.hardsoftdouble.HardSoftDoubleScore;
+import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
+import org.optaplanner.core.api.score.buildin.simple.SimpleScore;
+import org.optaplanner.core.api.score.buildin.simplebigdecimal.SimpleBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.simpledouble.SimpleDoubleScore;
+import org.optaplanner.core.api.score.buildin.simplelong.SimpleLongScore;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XStreamConverter(value = ToAttributedValueConverter.class, strings = { "scoreClass" })
+public class ScoreWrapper {
+
+    @XmlAttribute(name = "scoreClass")
+    @XStreamAlias(value = "scoreClass")
+    private Class<? extends Score> scoreClass;
+
+    @XmlValue
+    private String scoreString;
+
+    public ScoreWrapper() {
+    }
+
+    public Class<? extends Score> getScoreClass() {
+        return scoreClass;
+    }
+
+    public void setScoreClass( Class<? extends Score> scoreClass ) {
+        this.scoreClass = scoreClass;
+    }
+
+    public String getScoreString() {
+        return scoreString;
+    }
+
+    public void setScoreString( String scoreString ) {
+        this.scoreString = scoreString;
+    }
+
+    /**
+     * Returns score representation of the object.
+     *
+     * @return Score representation of the object.
+     * @throws UnsupportedOperationException If <code>scoreClass</code> is not one of the out-of-box score implementations. In this case
+     *                                       clients may implement their own way to extract the score object.
+     */
+    public Score toScore() {
+        if ( scoreClass == null ) {
+            return null;
+        }
+
+        if ( BendableScore.class.equals( scoreClass ) ) {
```

Comment:
Note that one might become missing once I add an new one upstream and forgot it here. At some point we'll add a HardMediumSoft one as there's one missing.

We should probably push these code into optaplanner-core. I am thinking a static method on ScoreDefinition or ScoreUtil or so.